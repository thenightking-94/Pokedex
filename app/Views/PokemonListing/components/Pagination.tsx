import { Grid } from '@mui/material';
import React, { ReactElement } from 'react';
import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';

export default function Pagination (props): ReactElement{
    const {data, pageNumber, goToNextPage, goToPrevPage} = props
    return(
        <Grid container className='pagination' alignItems={'center'} justifyContent='center' >
                 {data?.previous !== null && <ArrowBackIosNewIcon className='arrow' onClick={goToPrevPage}/>}
                  <p className='page-number'>{pageNumber}</p> 
                 {data?.next !== null && <ArrowForwardIosIcon className='arrow' onClick={goToNextPage}/>}
        </Grid>
    )

}