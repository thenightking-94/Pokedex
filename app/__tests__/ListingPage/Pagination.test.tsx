import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import Pagination from '../../Views/PokemonListing/components/Pagination'

const queryClient = new QueryClient()

describe('Pagination', () => {
  it('renders correctly', () => {
    render(
    <QueryClientProvider client={queryClient}>
         <Pagination/>
    </QueryClientProvider>
   )
  })
})

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <Pagination/>
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });