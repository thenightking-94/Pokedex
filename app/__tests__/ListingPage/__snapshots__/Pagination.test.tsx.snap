// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container pagination css-go2vvz-MuiGrid-root"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow css-i4bv87-MuiSvgIcon-root"
    data-testid="ArrowBackIosNewIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
    />
  </svg>
  <p
    className="page-number"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow css-i4bv87-MuiSvgIcon-root"
    data-testid="ArrowForwardIosIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
    />
  </svg>
</div>
`;
